pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -228,14 +237,47 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		}
 	} else {
 		if contains(instance.ObjectMeta.Finalizers, FinalizerName) {
-			for _, f := range []func(c client.Client, cr *hcov1alpha1.HyperConverged) error{
+			for i, f := range []func(c client.Client, cr *hcov1alpha1.HyperConverged, requestNamespace string) error{
+				// Keep ensuring that we are trying to delete
+				// protected resources first
+				ensureKubeVirtDeleted,
 				ensureCDIDeleted,
 				ensureNetworkAddonsDeleted,
 				ensureKubeVirtCommonTemplateBundleDeleted,
 			} {
-				err = f(r.client, instance)
+				err = f(r.client, instance, request.Namespace)
 				if err != nil {
 					reqLogger.Error(err, "Failed to manually delete objects")
+
+					errT := ErrHCOUninstall
+					errMsg := uninstallHCOErrorMsg
+					switch i {
```

Comment:
I think you should add a TODO to handle the case where a `Delete` fails that isn't because KubeVirt|CDI refused our delete request.

Reply:
in that case we are already going to emit a generic error message:
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/485/commits/9d30b92e0a22096e5f16b77572b236ac4a36315e#diff-64e1e2b9251fa5137c91d5a7c8427b88R70

Reply:
My issue is that it is possible that our call to `ensureKubeVirtDeleted` or `ensureCDIDeleted` to return an err and it not be because KubeVirt or CDI refused our request.

Reply:
now I got your point, thanks